networker/networker.go

```diff
@@ -1257,6 +1298,10 @@ func (netw *Combined) defaultMeshUnBlock() error {
 func (netw *Combined) UnSetMesh() error {
 	netw.mu.Lock()
 	defer netw.mu.Unlock()
+	// clear this flag only when user turns mesh off, cannot do that in internal func
+	// because it is used during refresh, and during refresh we need to know what
```

Comment:
Maybe I missed something but I don't see refresh case being handled at the moment.

Reply:
Sorry, comment is incorrect about the source of refreshVPN trigger - it's not NC event, it's network change event, `netw.routeToRemoteLan` we have state which was set during `mesh peer connect`, we keep this state for `refreshVPN` case (https://github.com/NordSecurity/nordvpn-linux/blob/3bd6de0bbccd1648fcdbed6eab0b6b98817631a4/networker/vpn.go#L35) which is invoked on network state change (https://github.com/NordSecurity/nordvpn-linux/blob/3bd6de0bbccd1648fcdbed6eab0b6b98817631a4/networker/listener.go#L10)

Reply:
Got it. So some other points:

1. Does it really need to be reset here? I mean it will be set anyways when a new connect will be done.
2. I think the case where the peer changes `lan-access` permission while we are routing through them is still unhandled.

Reply:
1. enableLocalTraffic state to keep is needed for refreshVPN during network change when we are mesh peer connected;
2. because policy router SetupRoutingRules is used during `mesh on` and `mesh peer connect` and also during those bothactions this is state is used, wee need to reset it to default value `true` when unseting mesh (by user specifically, not by internal func which is used for refreshVPN case).
3. well, we do not handle such cases when peer changes permissions (e.g. do we brake file transfer if peer changes that permission?), probably need new task for this;